install.py

```diff
@@ -0,0 +1,188 @@
+# License.
+
+# Python 2/3 compatibility
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import platform
+import shutil
+import sys
+from subprocess import check_call
+from os.path import join, abspath, isdir, dirname
+
+_PYPY = hasattr(sys, 'pypy_translation_info')
+_OS = platform.system()
+_WIN = _OS == 'Windows'
+_MAC = _OS == 'Darwin'
+_LINUX = _OS == 'Linux'
+
+_C_EXT_DEPENDENCY_DIR = abspath(join(dirname(os.path.abspath(__file__)), 'ion-c-build'))
+_C_EXT_DEPENDENCY_LIB_LOCATION = abspath(join(_C_EXT_DEPENDENCY_DIR, 'lib'))
+_C_EXT_DEPENDENCY_INCLUDES_LOCATIONS = abspath(join(_C_EXT_DEPENDENCY_DIR, 'include'))
+
+_IONC_REPO_URL = "https://github.com/amzn/ion-c.git"
+_IONC_DIR = abspath(join(dirname(os.path.abspath(__file__)), 'ion-c'))
+_IONC_LOCATION = abspath(join(dirname(os.path.abspath(__file__)), 'ion-c', 'build', 'release'))
+_IONC_INCLUDES_LOCATIONS = {
+    'ionc': abspath(join(dirname(os.path.abspath(__file__)), 'ion-c', 'ionc', 'include', 'ionc')),
+    'decNumber': abspath(join(dirname(os.path.abspath(__file__)), 'ion-c', 'decNumber', 'include', 'decNumber'))
+}
+
+_LIB_PREFIX = 'lib'
+
+_LIB_SUFFIX_MAC = '.dylib'
+_LIB_SUFFIX_WIN = '.lib'
+_LIB_SUFFIX_LINUX = '.so'
+
+
+def _get_lib_name(name):
+    if _MAC:
+        return '%s%s%s' % (_LIB_PREFIX, name, _LIB_SUFFIX_MAC)
+    elif _LINUX:
+        return '%s%s%s' % (_LIB_PREFIX, name, _LIB_SUFFIX_LINUX)
+    elif _WIN:
+        return '%s%s' % (name, _LIB_SUFFIX_WIN)
+
+
+def _library_exists():
+    return _library_exists_helper('ionc') and _library_exists_helper('decNumber')
+
+
+def _library_exists_helper(name):
+    return os.path.exists(join(_C_EXT_DEPENDENCY_LIB_LOCATION, _get_lib_name(name)))
+
+
+def _download_ionc():
+    try:
+        # Create a directory to store build output
+        if not isdir(_C_EXT_DEPENDENCY_DIR):
+            os.mkdir(_C_EXT_DEPENDENCY_DIR)
+            os.mkdir(_C_EXT_DEPENDENCY_LIB_LOCATION)
+            os.mkdir(_C_EXT_DEPENDENCY_INCLUDES_LOCATIONS)
+
+        # Install ion-c
+        if not isdir('./ion-c'):
+            check_call(['git', 'clone', '--recurse-submodules', _IONC_REPO_URL, 'ion-c'])
+        os.chdir('ion-c/')
+
+        # Initialize submodule
+        check_call(['git', 'submodule', 'update', '--init'])
+
+        # Build ion-c
+        _build_ionc()
+
+        os.chdir('../')
+        move_build_lib_for_distribution()
+    except:
+        if isdir(_C_EXT_DEPENDENCY_DIR):
+            shutil.rmtree(_C_EXT_DEPENDENCY_DIR)
+        if isdir(_IONC_DIR):
+            shutil.rmtree(_IONC_DIR)
+        print('ionc build error: Unable to build ion-c library.')
+        return False
+
+
+def _build_ionc():
+    if _WIN:
+        _build_ionc_win()
+    elif _MAC or _LINUX:
+        _build_ionc_mac_and_linux()
+
+
+def _build_ionc_win():
+    # check_call('cmake -G \"Visual Studio 15 2017 Win64\"')
+    check_call('cmake -G \"Visual Studio 16 2019\"')
```

Comment:
Need to figure out a regular way to compile ionc.

If we are going to leverage GitHub actions to distribute, need to check if the virtual machine GitHub hosts already installed VisualStudio.

Reply:
+1

Reply:
Will work on this in the "releasing beta stage" work.